src/main/java/com/me4502/modularframework/module/Module.java

```diff
@@ -34,12 +34,30 @@
 public @interface Module {
 
     /**
+     * The id of this module.
+     * @return The id of this module
+     */
+    String moduleId();
+
+    /**
      * The name of this module.
      * @return The name of this module
      */
     String moduleName();
 
     /**
+     * The version of this module.
+     * @return The version of this module
+     */
+    String moduleVersion();
```

Comment:
These should all have defaults.

I'd recommend 1.0.0 for version, empty array for authors, and redirect ID to moduleName().toLowerCase